main.go

```diff
@@ -124,118 +132,174 @@ func main() {
 
 	// If driverName specified, access source DB via database/sql
 	// driver. Otherwise read pgdump data from stdin.
-	fromPgDump = driverName == ""
-	if fromPgDump {
-		err = pgDumpToSpanner(project, instance, dbName, nil, filePrefix, now)
-	} else {
-		err = sourceToSpanner(driverName, project, instance, dbName, filePrefix, now)
+	if driverName == "" {
+		driverName = PGDUMP
 	}
-
+	err = toSpanner(driverName, project, instance, dbName, ioHelper, filePrefix, now)
 	if err != nil {
 		panic(err)
 	}
 }
 
-// sourceToSpanner migrates a source DB (accessed via a database/sql
-// driver) to Spanner.  The source DB must support the
-// information_schema standard (for information about tables, views,
-// columns etc). Data is accessed via SELECT statements.
-// TODO: add data conversion.
-func sourceToSpanner(driver, projectID, instanceID, dbName string, outputFilePrefix string, now time.Time) error {
-	fmt.Println(`###################################################################
-Accessing a source DB via an database/sql driver is an experimental
-feature that is not fully implemented. Currently we only do schema
-conversion and we only support PostgreSQL.
-###################################################################`)
-	if driver != "postgres" {
-		return fmt.Errorf("Driver %s not supported", driver)
+// toSpanner is the main entrance of the entire conversion, which runs the
+// following steps:
+//   1. Run schema conversion
+//   2. Create database
+//   3. Run data conversion
+//   4. Generate report
+func toSpanner(driver, projectID, instanceID, dbName string, ioHelper *ioStreams, outputFilePrefix string, now time.Time) error {
+	conv, err := schemaConv(driver, ioHelper)
+	if err != nil {
+		return err
+	}
+	// close the seekable file
+	if ioHelper.seekableIn != nil {
+		defer ioHelper.in.Close()
+	}
+
+	writeSchemaFile(conv, now, outputFilePrefix+schemaFile, ioHelper.out)
+	db, err := createDatabase(projectID, instanceID, dbName, conv, ioHelper.out)
+	if err != nil {
+		fmt.Printf("\nCan't create database: %v\n", err)
+		return fmt.Errorf("can't create database")
+	}
+
+	client, err := getClient(db)
+	if err != nil {
+		fmt.Printf("\nCan't create client for db %s: %v\n", db, err)
+		return fmt.Errorf("can't create Spanner client")
 	}
+
+	bw, err := dataConv(driver, ioHelper, client, conv)
+	if err != nil {
+		fmt.Printf("\nCan't finish data conversion for db %s: %v\n", db, err)
+		return fmt.Errorf("can't finish data conversion")
+	}
+	banner := getBanner(now, db)
+	report(bw.DroppedRowsByTable(), 0, banner, conv, outputFilePrefix+reportFile, ioHelper.out)
+	writeBadData(bw, conv, banner, outputFilePrefix+badDataFile, ioHelper.out)
+	return nil
+}
+
+func schemaConv(driver string, ioHelper *ioStreams) (*internal.Conv, error) {
+	switch driver {
+	case POSTGRES:
+		return schemaConvFromSourceDB(driver)
```

Comment:
This is kind of a long name for go.
How about just schemaFromSQL?
And schemaFromPgDump instead of schemaConvFromFileStream.